### PR TITLE
openlineage, trino: add OpenLineage support for Trino.

### PR DESCRIPTION
**Pull request by JDarDagran**
This PR adds OpenLineage support for TrinoOperator.

### CODE REVIEW

File: airflow/providers/openlineage/utils/sql.py [hunk 1]
```diff
@@ -155,11 +155,17 @@ def create_information_schema_query(
     metadata = MetaData(sqlalchemy_engine)
     select_statements = []
     for db, schema_mapping in tables_hierarchy.items():
-        schema, table_name = information_schema_table_name.split(".")
         if db:
-            schema = f"{db}.{schema}"
+            schema = db
```
Comment (mobuchowski) [inline airflow/providers/openlineage/utils/sql.py:163]: Does this mean `schema` is `database schema` in Oracle meaning here, not `schema of a database`? Would be nice to document this if true - this is cause of many errors usually.

File: airflow/providers/trino/hooks/trino.py [hunk 2]
```diff
@@ -233,3 +233,32 @@ def _serialize_cell(cell: Any, conn: Connection | None = None) -> Any:
...
+            database=connection.extra_dejson.get("catalog", "hive"),
+            is_information_schema_cross_db=True,
+        )
+
+    def get_openlineage_database_dialect(self, _):
+        """Returns database dialect."""
+        return "trino"
+
+    def get_openlineage_default_schema(self):
+        """MySQL has no concept of schema."""
```
Comment (mobuchowski) [inline airflow/providers/trino/hooks/trino.py]:
```suggestion
        """Trino has no concept of schema."""
```

File: airflow/providers/openlineage/utils/sql.py (hunk 1 above)
Comment (JDarDagran) [reply airflow/providers/openlineage/utils/sql.py:163]: That is a technical solution related to how SQLAlchemy constructs table identifiers. The assumption is to have `database` -> `schema` -> `table_name` hierarchy with first two optional.
I've added a short explanation how above solution works.

Comment (mobuchowski) [reply airflow/providers/openlineage/utils/sql.py:163]: 👍

**Review by mobuchowski — APPROVED** [review]

**Review by potiuk — APPROVED** [review]
Flake only. Merging.

**Pull request merged**